File: tests/src/Entity/StringValuesEnum.php
<?php

namespace Dayploy\DartDtoBundle\Tests\src\Entity;

use Dayploy\DartDtoBundle\Attributes\DartDto;

#[DartDto]
enum StringValuesEnum: string
{
    case none = 'none';
    case warning = 'warning';
    case error = 'error';
}

File: tests/src/entity/string_values_enum.dart
enum StringValuesEnum {
  none("none"),
  warning("warning"),
  error("error");

  final String value;
  const StringValuesEnum(this.value);

  factory StringValuesEnum.fromValue(String value) {
    return StringValuesEnum.values.firstWhere((e) => e.value == value);
  }
}
